File: viewpagerwithoutfragment/src/main/java/com/kirchhoff/viewpagerwithoutfragment/CustomPagerEnum.java
package com.kirchhoff.viewpagerwithoutfragment;

import android.support.annotation.LayoutRes;
import android.support.annotation.StringRes;

/**
 * @author Kirchhoff-
 */

public enum CustomPagerEnum {

    RED(R.string.red, R.layout.v_red),
    BLUE(R.string.blue, R.layout.v_blue),
    ORANGE(R.string.orange, R.layout.v_orange);

    @StringRes
    private int titleResId;
    @LayoutRes
    private int layoutResId;

    CustomPagerEnum(@StringRes int titleResId, @LayoutRes int layoutResId) {
        this.titleResId = titleResId;
        this.layoutResId = layoutResId;
    }

    public int getTitleResId() {
        return titleResId;
    }

    public int getLayoutResId() {
        return layoutResId;
    }
}

File: viewpagerwithoutfragment/src/main/java/com/kirchhoff/viewpagerwithoutfragment/CustomPagerEnum.kt
package com.kirchhoff.viewpagerwithoutfragment

import androidx.annotation.LayoutRes
import androidx.annotation.StringRes

enum class CustomPagerEnum(
        @param:StringRes @field:StringRes val titleResId: Int,
        @param:LayoutRes @field:LayoutRes val layoutResId: Int
) {
    RED(R.string.red, R.layout.v_red),
    BLUE(R.string.blue, R.layout.v_blue),
    ORANGE(R.string.orange, R.layout.v_orange)
}
